### DISCUSSION TITLE
Reference to src being replaced when ChangeTurf() is called

### DISCUSSION THREAD

**Discussion by SteelSlayer** [ParadiseSS13/Paradise#15888] [Code Help] [answered]
Posted this on discord but posting again here so it won't get lost as easily.

Given a list which currently contains a reference to the `src` floor:  whenever `/turf/proc/ChangeTurf` is called, it qdeletes the `src` floor and creates a new plating. 

Somehow, during the `new`ing of the plating, the reference to the `src` floor in the list gets *replaced* with the plating. I have no idea whats going on here. I see no code that should do this, so I'm guessing its some behind the scenes Byond stuff.

Gif for a visual explanation:

![vT25PFS](https://user-images.githubusercontent.com/42044220/114764330-7067c180-9d29-11eb-8615-e852f91083bf.gif)

I haven't modified any existing turf related code on my branch, and in fact, this issue happens on master as well with no code additions.

**Comment by farie82** [comment, marked as answer]
http://www.byond.com/docs/ref/#/turf
There it suggests that if you new a turf with as param the old one that it will replace the old one.
> Turfs cannot be moved. They can only be created or destroyed by changing world.maxx, world.maxy, or world.maxz. When you create a new turf with new(), it always replaces the old one.

You also can't move or new/delete them normally. So it makes sense that byond will reuse the reference

**Comment by Fox-McCloud** [comment]
What @farie82 said is correct. You can't really "delete" a turf, per se, only replace an instance of it; this is why calling `del` on a turf isn't really that much more expensive than `qdel`.

Areas, IIRC, function pretty much the exact same way.

**Comment by SteelSlayer** [comment]
Yep, that checks out, thanks guys.
